hilbert/solver.py

```diff
@@ -28,20 +30,26 @@ def __init__(
         that go within the core training loop.  It lets callers of 
         Solver.cycle() concisely ask to iterate forward by some number
         of updates, without needing to know all the pieces of machinery in the
-        training loop and how they work together
+        training loop and how they work together.
+
+        Solver.cycle() also support gradient accumulation calculation.
         """
 
         # Own it like you do
         self.loader = loader
         self.loss = loss
         self.optimizer = optimizer
         self.learner = learner
-        self.schedulers = schedulers or []
+        self.schedulers = schedulers
```

Comment:
I think we need to revert this.  If we don't have any schedulers, then it should be `[]` not None, since attempting to iterate over `None` will error.
```suggestion
        self.schedulers = schedulers or []
```